src/Compilers/Core/Portable/CodeGen/VariableSlotAllocator.cs

```diff
@@ -79,5 +80,24 @@ internal abstract class VariableSlotAllocator
         /// or lambda body syntax (<paramref name="isLambdaBody"/> is true).
         /// </summary>
         public abstract bool TryGetPreviousLambda(SyntaxNode lambdaOrLambdaBodySyntax, bool isLambdaBody, out DebugId lambdaId);
+
+        /// <summary>
+        /// State number to be used for next state of the state machine,
+        /// or <see langword="null"/> if none of the previous versions of the method was a state machine with a increasing state
+        /// </summary>
+        /// <param name="increasing">True if the state number increases with progress, false if it decreases.</param>
```

Comment:
>True if the state number increases with progress, false if it decreases. [](http://example.com/codeflow?start=37&length=72)

nit: Consider "Indicates whether the first unused increasing or decreasing state should be returned. Decreasing states are used for disposal states (suspensions within finally blocks)."

Reply:
Decreasing states can be used for two different purposes:
1) in iterators it's the disposal states that correspond to finally blocks
2) in async iterators they are the iterator states (disposal is handled differently), increasing states are async states